webui/react/src/components/PaginatedNotesCard.tsx

```diff
@@ -0,0 +1,203 @@
+    setDeleteTarget(deletePageNumber);
+  }, [ onDelete, setDeleteTarget ]);
+
+  const handleEditedNotes = useCallback((newContents: string) => {
+    setEditedContents(newContents);
+  }, []);
+
+  useEffect(() => {
+    if (notes.length === 0) return;
+    if (currentPage < 0) { setCurrentPage(0); fireNoteChangeSignal(); }
+    if (currentPage >= notes.length) { setCurrentPage(notes.length - 1); fireNoteChangeSignal(); }
+  }, [ currentPage, notes.length, fireNoteChangeSignal ]);
+
+  useEffect(() => {
+    setEditedContents(notes?.[currentPage]?.contents ?? '');
+    setEditedName(notes?.[currentPage]?.name ?? '');
+  }, [ currentPage, notes ]);
+
+  const ActionMenu = useCallback((pageNumber: number) => {
+    return (
+      <Menu onClick={({ domEvent }) => domEvent.stopPropagation()}>
+        <Menu.Item
+          danger
+          key="delete"
+          onClick={() => handleDeletePage(pageNumber)}>
+          Delete...
+        </Menu.Item>
+      </Menu>
+    );
+  }, [ handleDeletePage ]);
+
+  if (notes.length === 0) {
+    return (
+      <div className={css.emptyBase}>
+        <div className={css.messageContainer}>
+          <Icon name="document" size="mega" />
+          <p>No notes for this project</p>
+          <Button onClick={handleNewPage}>+ New Page</Button>
+        </div>
+      </div>
+    );
+  }
+
+  return (
+    <div className={css.base}>
+      {notes.length > 0 && (
+        <div className={css.sidebar}>
+          <ul className={css.listContainer} role="list">
+            {(notes as Note[]).map((note, idx) => (
+              <Dropdown
+                disabled={disabled}
+                key={idx}
+                overlay={() => ActionMenu(idx)}
+                trigger={[ 'contextMenu' ]}>
+                <li
+                  className={css.listItem}
+                  style={{
+                    borderColor: idx === currentPage ?
+                      'var(--theme-colors-monochrome-12)' :
+                      undefined,
+                  }}
+                  onClick={() => handleSwitchPage(idx)}>
+                  <span>{note.name}</span>
+                  {!disabled && (
+                    <Dropdown
+                      overlay={() => ActionMenu(idx)}
+                      trigger={[ 'click' ]}>
+                      <div className={css.action} onClick={e => e.stopPropagation()}>
+                        <Icon name="overflow-horizontal" />
+                      </div>
+                    </Dropdown>
+                  )}
+                </li>
+              </Dropdown>
+            ))}
+          </ul>
+        </div>
+      )}
+      <div className={css.notesContainer}>
+        <NotesCard
+          disabled={disabled}
+          extra={(
+            <Dropdown
+              overlay={() => ActionMenu(currentPage)}
+              trigger={[ 'click' ]}>
+              <div style={{ cursor: 'pointer' }}>
+                <Icon name="overflow-horizontal" />
+              </div>
```

Comment:
**nit:** use className on `Icon` directly to shorten this to one line? For example:
```
<Icon className={css.pointer} name="overflow-horizontal" />
```

Reply:
Similar to the tooltip previously, the `div` is necessary for the Dropdown to attach to.